ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/StatePointers.kt

```diff
@@ -0,0 +1,347 @@
+package net.corda.v5.ledger.utxo
+
+import net.corda.v5.base.annotations.CordaSerializable
+import java.util.*
+
+/**
+ * Defines a pointer to a [ContractState].
+ *
+ * @property identifier The identifier of the [ContractState] being pointed to.
+ * @property type The type of the [ContractState] being pointed to.
+ * @property isResolved Determines whether the current pointer should be resolved to a reference input when used in a transaction.
+ */
+@CordaSerializable
+interface StatePointer<T : ContractState> {
+    val identifier: Any
+    val type: Class<T>
+    val isResolved: Boolean
+}
+
+/**
+ * Represents a state pointer that points statically to a specific [StateRef].
+ *
+ * @constructor Creates a new instance of the [StaticPointer] data class.
+ * @property identifier The identifier of the [ContractState] being pointed to.
+ * @property type The type of the [ContractState] being pointed to.
+ * @property isResolved Determines whether the current pointer should be resolved to a reference input when used in a transaction.
+ */
+data class StaticPointer<T : ContractState>(
```

Comment:
With the amount of code in these classes, they should be split into separate files with a appropriate package.